mdoc/Mono.Documentation/Updater/AttributeParserContext.cs

```diff
@@ -99,16 +107,36 @@ private void ReadNullableAttribute()
         }
 
         private void ReadTupleElementNames()
+        {
+            tupleElementNames = ReadCustomAttributeValue<string>(
+                null,
+                Consts.TupleElementNamesAttribute);
+        }
+
+        private void ReadNativeIntegerAttribute()
+        {
+            nativeIntegerFlags = ReadCustomAttributeValue<bool>(
+                () => new bool[] { true },
+                Consts.NativeIntegerAttribute);
+        }
+
+        private T[] ReadCustomAttributeValue<T>(Func<T[]> init, string attributeName)
         {
             if (provider != null && provider.HasCustomAttributes)
```

Comment:
if (provider == null || !provider.HasCustomAttributes) return null; ?

How about return null first instead of a big if code block?

Reply:
updated